### PR TITLE
Refactor HDF5 handlers and add DASK handler

### PR DESCRIPTION
**Pull request by stuwilkins**
This is an attempt to refactor the HDF5 handlers and add handlers for returning DASK objects

### CODE REVIEW

**Comment by stuwilkins** [comment]
@danielballan @tacaswell @stuartcampbell 

This is now ready for review. I have tried to restructure the HDF5 stuff. 

The question is what to do with autoregister, one possibility is what is here. I was also considering the possibility of having a member of the class be something like `autoregister=True` to configure what comes out of the box. Or no autoregister and configure this in the yaml config files for all beamlines. Thoughts? I am happy to do the work!

File: databroker/assets/handlers.py [hunk 1]
```diff
@@ -6,54 +6,13 @@
...
+import dask.array as da
+
 from .handlers_base import HandlerBase
 from .readers.spe import PrincetonSPEFile
-from pims import FramesSequence, Frame
 
 logger = logging.getLogger(__name__)
 
-
-# The ImageCube class is used for a per event representation of
```
Comment (tacaswell) [inline databroker/assets/handlers.py]: Why did these get deleted?

**Comment by stuwilkins** [comment]
@tacaswell Sorry, over aggressive refactoring, I was intending on dropping PIMS but that is probably not a good idea ... I will add an HDF5 Handler to just return the h5py object

**Comment by stuwilkins** [comment]
@danielballan done... travis fails though (not on the handlers)

**Comment by danielballan** [comment]
The failure is due to pims. It has been fixed and released upstream, but the release has not yet made its way through conda-forge.

**Pull request merged**